diff --git a/media/libstagefright/MediaCodecList.cpp b/media/libstagefright/MediaCodecList.cpp
index c97a3a9..c319f7b 100644
--- a/media/libstagefright/MediaCodecList.cpp
+++ b/media/libstagefright/MediaCodecList.cpp
@@ -26,6 +26,8 @@
 #include <media/stagefright/OMXCodec.h>
 #include <utils/threads.h>
 
+#include <cutils/properties.h>
+
 #include <libexpat/expat.h>
 #ifdef QCOM_HARDWARE
 #include "include/ExtendedUtils.h"
@@ -445,6 +447,17 @@ ssize_t MediaCodecList::findCodecByType(
 }
 
 ssize_t MediaCodecList::findCodecByName(const char *name) const {
+
+
+    char prop_retrieved_value[PROPERTY_VALUE_MAX];
+
+    int ret = property_get("vu.co.meticulus.h264switch", prop_retrieved_value, "false");
+
+    if(strcmp(name, "OMX.ST.VFM.H264Dec") == 0 && strcmp(prop_retrieved_value, "true") == 0){
+	ALOGI("h264switch: %s - %s -> OMX.google.h264.decoder",__func__,name);
+	name = "OMX.google.h264.decoder";
+    }
+
     for (size_t i = 0; i < mCodecInfos.size(); ++i) {
         const CodecInfo &info = mCodecInfos.itemAt(i);
 
@@ -464,8 +477,19 @@ const char *MediaCodecList::getCodecName(size_t index) const {
     if (index >= mCodecInfos.size()) {
         return NULL;
     }
+    const CodecInfo &binfo = mCodecInfos.itemAt(index);
+
+    char prop_retrieved_value[PROPERTY_VALUE_MAX];
+
+    int ret = property_get("vu.co.meticulus.h264switch", prop_retrieved_value, "false");
+
+    if(strcmp(binfo.mName.c_str(), "OMX.ST.VFM.H264Dec") == 0 && strcmp(prop_retrieved_value, "true") == 0){
+	ALOGI("h264switch: %s - %s -> OMX.google.h264.decoder",__func__,binfo.mName.c_str());
+	index = findCodecByName("OMX.google.h264.decoder");
+    }
 
     const CodecInfo &info = mCodecInfos.itemAt(index);
+
     return info.mName.c_str();
 }
 
@@ -484,6 +508,17 @@ bool MediaCodecList::codecHasQuirk(
         return NULL;
     }
 
+    const CodecInfo &binfo = mCodecInfos.itemAt(index);
+
+    char prop_retrieved_value[PROPERTY_VALUE_MAX];
+
+    int ret = property_get("vu.co.meticulus.h264switch", prop_retrieved_value, "false");
+
+    if(strcmp(binfo.mName.c_str(), "OMX.ST.VFM.H264Dec") == 0 && strcmp(prop_retrieved_value, "true") == 0){
+	ALOGI("h264switch: %s - %s -> OMX.google.h264.decoder",__func__,binfo.mName.c_str());
+	index = findCodecByName("OMX.google.h264.decoder");
+    }
+
     const CodecInfo &info = mCodecInfos.itemAt(index);
 
     if (info.mQuirks != 0) {
@@ -504,6 +539,17 @@ status_t MediaCodecList::getSupportedTypes(
         return -ERANGE;
     }
 
+    const CodecInfo &binfo = mCodecInfos.itemAt(index);
+
+    char prop_retrieved_value[PROPERTY_VALUE_MAX];
+
+    int ret = property_get("vu.co.meticulus.h264switch", prop_retrieved_value, "false");
+
+    if(strcmp(binfo.mName.c_str(), "OMX.ST.VFM.H264Dec") == 0 && strcmp(prop_retrieved_value, "true") == 0){
+	ALOGI("h264switch: %s - %s -> OMX.google.h264.decoder",__func__,binfo.mName.c_str());
+	index = findCodecByName("OMX.google.h264.decoder");
+    }
+
     const CodecInfo &info = mCodecInfos.itemAt(index);
 
     for (size_t i = 0; i < mTypes.size(); ++i) {
@@ -529,6 +575,18 @@ status_t MediaCodecList::getCodecCapabilities(
         return -ERANGE;
     }
 
+    const CodecInfo &binfo = mCodecInfos.itemAt(index);
+
+    char prop_retrieved_value[PROPERTY_VALUE_MAX];
+
+    int ret = property_get("vu.co.meticulus.h264switch", prop_retrieved_value, "false");
+
+    if(strcmp(binfo.mName.c_str(), "OMX.ST.VFM.H264Dec") == 0 && strcmp(prop_retrieved_value, "true") == 0){
+	ALOGI("h264switch: %s - %s -> OMX.google.h264.decoder",__func__,binfo.mName.c_str());
+	index = findCodecByName("OMX.google.h264.decoder");
+	
+    }
+
     const CodecInfo &info = mCodecInfos.itemAt(index);
 
     OMXClient client;
